MCLOUD-6073: Review "Update the metapackage" devdocs which is confusing merchants and support agents (#8236)
magento/devdocs: src/cloud/project/ece-tools-upgrade-project.md

>=current_version <next_version
```

> where `current_version` is the version of Magento you want to be installed
> 
> and `next_version` is the next patch version after your `current_version`.

Reviewer: ```suggestion
-  For `current_version`, specify the Magento version to install.
-  For `next_version`, specify the next patch version after the value specified in `current_version`.
```

> If you want to install Magento `2.3.5-p2` you still should set `current_version` as `2.3.5` and constraint `">=2.3.5 <2.3.6"` will install the latest available package for 2.3.5

Author: @BarnyShergold Sometimes, we have `-px` releases even in cloud metapackages, that is why we still recommend to use that constraint. Probably, we need to mention this in documentation.

> If you want to install Magento `2.3.5-p2` you still should set `current_version` as `2.3.5` and constraint `">=2.3.5 <2.3.6"` will install the latest available package for 2.3.5

Reviewer: ```suggestion
If you want to install Magento `2.3.5-p2`, you still set `current_version` to `2.3.5` and the `next_version` to `2.3.6`. The constraint `">=2.3.5 <2.3.6"` installs the latest available package for 2.3.5.
```

> If you want to install Magento `2.3.5-p2` you still should set `current_version` as `2.3.5` and constraint `">=2.3.5 <2.3.6"` will install the latest available package for 2.3.5

Reviewer 2: Yes, but as this is in the Metapackage - ">=2.3.4 <2.3.5" it would pick up the latest -px release and install?

> If you want to install Magento `2.3.5-p2` you still should set `current_version` as `2.3.5` and constraint `">=2.3.5 <2.3.6"` will install the latest available package for 2.3.5

Author: @BarnyShergold 
sorry, did not get your question.
if you set `"magento/magento-cloud-metapackage": "2.3.5"` in composer.json but we released 2.3.5-p1 version - you still will have correct version of EE, but you won't have necessary changes in `magento/magento-cloud-metapackage`. E.g. we may change constraint for other packages like `magento/module-paypal-on-boarding` or add new package. That is why we strongly recommend to use constraint as `"magento/magento-cloud-metapackage": ">=2.3.5 <2.3.6"`
